### PR TITLE
Fix order detail total amount update when rounding is per item

### LINKED ISSUE #21717: BO - Order page - An exception is displayed when changing the invoice address in the second time - Round mode = Round on each item
#### Describe the bug

An exception is displayed when changing the invoice address the second time.

#### Expected behavior

Explain what you expected to happen instead.

#### Steps to Reproduce

1. Download the United States pack from BO > International > Localization page
2. In the BO > Shop Parameters > General > set Round type = **Round on each item**
3. Create a Tax with 21% for France country
4. Create a product with Tax excluded = 7.8 euro / Tax = 21%
5. Go to FO > Add the product to the cart with quantity = 80
6. Continue checkout with `pub@prestashop.com` customer
7. In the Address step, select "My Address" (United States)
8. Place the order
9. Go to BO > Order page > view last order created
10. Change the Shipping address to "Mon addresse" (Paris)
11. Change the Invoice address to "Mon addresse" (Paris)
12. Change invoice address to "My address" (United States)
13. see the error
image

PS: if you try to add a discount / add a product / update the product ordered => error
image

PS: 1768 => ok
#### Additional information

* PrestaShop version: 177x
* PHP version: 7.2

### PR DESCRIPTION
**Pull request by matthieu-rolland**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | In the `OrderAmountUpdater` service, the `total_price_tax_incl` amount per order detail was not calculated correctly
| Type?         | bug fix
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #21717
| How to test?  | See steps in issue #21717

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/21988)

### CODE REVIEW

**Review by Progi1984 — COMMENTED** [review]
Need a test (behat or unit) please

**Comment by matthieu-rolland** [comment]
> Need a test (behat or unit) please

yep it's coming !

**Review by Progi1984 — DISMISSED** [review]

File: tests/Integration/Behaviour/Features/Scenario/Order/order_shipping.feature [hunk 1]
```diff
@@ -850,3 +850,144 @@ Feature: Order from Back Office (BO)
     And the first invoice from order "bo_order1" should have following shipping tax details:
       | total_tax_excl | rate | total_amount |
       | 5.0            | 0.00 | 0.00         |
+
+  Scenario: In rounding per item mode, I use and address without taxes (no order_detail_tax created), then I change to a country with taxes all is correctly computed
```
Comment (atomiix) [inline tests/Integration/Behaviour/Features/Scenario/Order/order_shipping.feature]:
```suggestion
  Scenario: In rounding per item mode, I use an address without taxes (no order_detail_tax created), then I change to a country with taxes all is correctly computed
```

**Review by matks — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Comment by SD1982** [comment]
LGTM Thanks @matthieu-rolland !!

**Review by Progi1984 — CHANGES_REQUESTED** [review]
## DONT MERGE BEFORE 177X RELEASE

**Review by Progi1984 — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @matthieu-rolland & @SD1982

**Pull request merged**